shell/browser/api/message_port.cc

```diff
@@ -111,11 +119,14 @@ void MessagePort::Close() {
     gin_helper::EmitEvent(isolate, self, "close");
 }
 
-void MessagePort::Entangle(mojo::ScopedMessagePipeHandle handle) {
-  DCHECK(handle.is_valid());
+void MessagePort::Entangle(blink::MessagePortDescriptor port) {
+  DCHECK(port.IsValid());
   DCHECK(!connector_);
+  port_ = std::move(port);
   connector_ = std::make_unique<mojo::Connector>(
-      std::move(handle), mojo::Connector::SINGLE_THREADED_SEND,
+      port_.TakeHandleToEntangle(blink::ExecutionContext::From(
+          v8::Isolate::GetCurrent()->GetCurrentContext())),
```

Comment:
Can we use `TakeHandleToEntangleWithEmbedder` here and avoid using blink internals?

Reply:
Its handled here https://github.com/electron/electron/pull/23379/commits/f2ce4e8e047f0f0fe884a800bbf7a4beaa733444 :)

Reply:
brilliant.